### DISCUSSION TITLE
CheckNames parameter in BasedOnRow behaviour changed

### DISCUSSION THREAD

**Discussion by dfaruque** [serenity-is/Serenity#6222] [General]
`[BasedOnRow(typeof(Entities.OrderRow), CheckNames = true)]`
in previous version of serenity, if property names does not matched with the row then it was giving error on startup page of the application. but currently it is not giving the error at application startup instead it gives the error in a alert box when the Order page is visited. is it intended?

sample error msg:
`System.InvalidProgramException: Serene1.Northwind.Columns.OrderColumns has a [BasedOnRow(typeof(SerExtraNet5.Northwind.Entities.OrderRow), CheckNames = true)] attribute but its 'CustomerCompanyNamee' property doesn't have a matching field with same property / field name in the row. Please check if property is named correctly. To remove this validation you may set CheckNames to false on [BasedOnRow] attribute. To disable check for this specific property add a [IgnoreName] attribute to the property itself`

**Comment by volkanceylan** [comment]
There is no change about it, it did not raise error on startup afaik, it is a lazy script, so it will raise an error first time it is generated
